### PR TITLE
[Build] Added PYI files check

### PR DESCRIPTION
**Pull request by ilya-lavrenov**

### CODE REVIEW

File: src/python/CMakeLists.txt [hunk 1]
```diff
@@ -78,6 +85,67 @@ install(FILES "${OpenVINOGenAI_SOURCE_DIR}/LICENSE"
...
+                         RESULT_VAR pybind11_stubgen_AVAILABLE
+                         WARNING_MESSAGE "Please, install ${pybind11_stubgen_dep} if you plan to develop Python OpenVINO GenAI API"
+                         MESSAGE_MODE WARNING)
+elseif(DEFINED PY_BUILD_CMAKE_PACKAGE_NAME)
+    # in case of wheel build, pybind11-stubgen is always available via pyproject.toml's build-system
+    set(pybind11_stubgen_AVAILABLE ON)
+endif()
+
+if(pybind11_stubgen_AVAILABLE)
+    if(DEFINED ENV{CI} OR DEFINED ENV{TF_BUILD} OR DEFINED ENV{JENKINS_URL})
```
Comment (Wovchena) [inline src/python/CMakeLists.txt:116]: You need to set one of these in precommit build

File: src/python/CMakeLists.txt [hunk 2]
```diff
@@ -78,6 +85,67 @@ install(FILES "${OpenVINOGenAI_SOURCE_DIR}/LICENSE"
...
+                    -D source_pyi_files_location=${OpenVINOGenAI_SOURCE_DIR}/src/python
+                    -P "${CMAKE_CURRENT_SOURCE_DIR}/compare_pyi.cmake"
+        DEPENDS
+            ${python_sources}
+            "${CMAKE_CURRENT_SOURCE_DIR}/openvino_genai/__init__.py"
+            "${CMAKE_CURRENT_SOURCE_DIR}/compare_pyi.cmake"
+        COMMENT "[${pybind11_stubgen_dep}] Generate .pyi files"
+        VERBATIM)
+
+    add_custom_target(${TARGET_NAME}_stub ALL DEPENDS ${output_files})
```
Comment (Wovchena) [inline src/python/CMakeLists.txt:143]: py_openvino_genai_stub must always depend on src .pyi.

1. I run `export CI=1`
2. build passed
3. corrupted src `py_openvino_genai.pyi`
4. build failed
5. restored src `py_openvino_genai.pyi`
6. build failed with the same message

**Review by Wovchena — CHANGES_REQUESTED** [review]

File: src/python/CMakeLists.txt (hunk 1 above)
Comment (ilya-lavrenov) [reply src/python/CMakeLists.txt:119]: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#default-environment-variables

File: src/python/CMakeLists.txt (hunk 2 above)
Comment (ilya-lavrenov) [reply src/python/CMakeLists.txt:163]: please, check once again

File: requirements-build.txt [hunk 3]
```diff
@@ -1 +1,2 @@
-cmake~=3.30
\ No newline at end of file
+cmake~=3.23
+pybind11-stubgen==2.5.1
```
Comment (mryzhov) [inline requirements-build.txt:2]: looks like we don't need this file anymore, since all those requirements will be taken from toml

Comment (ilya-lavrenov) [reply requirements-build.txt:2]: it's used by Jenkins

**Review by Wovchena — APPROVED** [review]

**Pull request merged**